### ISSUE TITLE
Update README so build instructions are current and complete

### ISSUE THREAD

**Issue by julianharty** [kiwix/kiwix-android#112]
The current contents of the README are out of date. They describe how the app used to be build before being migrated to this repo (kiwix-android). I think the build is now simpler. Let's revise the README accordingly.

I'm basing my assessment on the travis config https://github.com/kiwix/kiwix-android/blob/master/.travis.yml

I'll update these notes with the build steps I take to build the main kiwix app & can then I/we can revise the README.

**Comment by julianharty** [comment]
Note: `./gradlew build` needs to know where Android tools are.
```
* What went wrong:
A problem occurred configuring project ':app'.
> SDK location not found. Define location with sdk.dir in the local.properties file or with an ANDROID_HOME environment variable.
```
Let's document this expectation & perhaps suggest how people can address it.

**Comment by julianharty** [comment]
`export ANDROID_HOME=/Users/julianharty/Library/Android/sdk`

```
Julians-MacBook-Pro:kiwix-android julianharty$ ./gradlew build
NDK is missing a "platforms" directory.
If you are using NDK, verify the ndk.dir is set to a valid NDK directory.  It is currently set to /Users/julianharty/Library/Android/sdk/ndk-bundle.
If you are not using NDK, unset the NDK variable from ANDROID_NDK_HOME or local.properties to remove this warning.
```
Do we need the NDK to build the app?

**Comment by mhutti1** [comment]
@julianharty no, I am pretty certain not. (Although I think you do if you are building from sources as is done in kiwix-build)

**Comment by mhutti1** [comment]
At the moment we use an old version of kiwix-lib at some point we need to look at importing the newest version automatically with gradle. I think the build should be possible just with ./gradle build if the android sdk is set. You can avoid this by importing into android studio and this will step you through this process.

**Comment by julianharty** [comment]
BUILD SUCCESSFUL

**Comment by julianharty** [comment]
From the vast set of tasks listed by `./gradlew tasks` I'd appreciate guidance on which targets to use when. I've read the task descriptions and they're too similar for me to guess which ones would run the Espresso tests, which one(s) would deploy a debug build of standard Kiwix on my device, etc.

**Comment by mhutti1** [comment]
Yeah that sounds like a good idea, basic rundown.the first word is the command:

- compile - Build an apk

- install/uninstall - As named

- connected - runs instrumentation (espresso) tests

- test - Runs unit tests

Some commands depend on others. I.e connected will compile if it needs too.

The next words specify what flavour/build variant to use. 

The standard is KiwixDebug

You join an action word with a build variant to get the full command.

**Comment by reeshabhkumarranjan** [comment]
I am getting the same error while building the app. Kindly provide details on how to solve this. I have tried some suggestions from the answers on [this StackOverflow question](https://stackoverflow.com/a/35702596/5394180). But they are not working.

**Comment by abdulwd** [comment]
@reeshabhkumarranjan Please try this https://github.com/kiwix/kiwix-android/pull/974#issuecomment-463157239.

**Comment by reeshabhkumarranjan** [comment]
@abdulwd Your comment seems promising. I am cloning the whole project again. The cloning of this project takes a lot of time, I don't know why. WIll update you on this soon.

**Comment by reeshabhkumarranjan** [comment]
I tried your fix, but it didn't work. I got these warnings:
```
WARNING: The option setting 'android.jetifier.blacklist=byte-buddy,adapter-rxjava2,converter-simplexml,logging-interceptor,mockwebserver,error_prone_annotations,guava' is experimental and unsupported.
Affected Modules: app, kiwixlib
```
And this:
```
WARNING: BuildType(local_download_server): buildConfigField 'KIWIX_DOWNLOAD_URL' value is being replaced: "http://download.kiwix.org/" -> "http://kiwix-download-server/"
Affected Modules: app
```
Also, the "Run" button is greyed out.
![image](https://user-images.githubusercontent.com/25811225/53867321-9afad280-4019-11e9-8b5d-e1dcb7cdcf5d.png)

**Comment by abdulwd** [comment]
Have you deleted the .idea folder before importing the project again? Also select kiwixDebug as the build variant.

**Comment by reeshabhkumarranjan** [comment]
Finally. Selecting `kiwixDebug` as build variant fixed the issue. This instruction should be in the README.md!

**Comment by abdulwd** [comment]
@reeshabhkumarranjan You can open a PR for that.

**Comment by mhutti1** [comment]
@reeshabhkumarranjan It is and is in bold:

> We utilize different build variants (flavours) to build various different versions of our app. Ensure your build variant is kiwixDebug to build the standard app.

**Comment by kelson42** [comment]
I think this ticket could be closed and has been supperseeded by #1200

**Comment by macgills** [comment]
I think we provide sufficient information, closing
